src/hashing_utils/scalar_storage.rs

```diff
@@ -0,0 +1,52 @@
+//! This module defines a Wrap-up over the dusk-bls12_381 to define it's kelvin
+//! storage traits
+
+use dusk_bls12_381::Scalar;
+use kelvin::{ByteHash, Content, Sink, Source};
+use std::borrow::Borrow;
+use std::io;
+use std::io::Read;
+
+#[derive(Debug, Clone)]
+/// This struct is a Wrapper type over the bls12-381 `Scalar` which has implemented
+/// inside the logic to allows `Kelvin` Merkle Trees understand how to store `Scalar`s
+/// inside of them leaves.
+///
+/// This Struct is the one that we will use inside of our SmartContract storage logic to
+/// encode/compress all of our Data Structures data into a single `Scalar`.
+pub struct StorageScalar(pub Scalar);
```

Comment:
Because this is what you're allowed to store inside Kelvin trees. You'll not be able to store `Scalar` directly.

Therefore I export it, so any library that deals with kelvin trees will need to implement the conversions from them custom types to this `StorageScalar`.